Apps/WebClient/src/ClientApp/src/app.vue

```diff
@@ -261,6 +262,10 @@ function currentPathMatches(...paths: string[]): boolean {
     return paths.some((path) => path === currentPath);
 }
 
+function releaseTicketBeforeUnloadListener(): void {
+    store.dispatch("waitlist/releaseTicket");
+}
```

Comment:
I'll simplfy the name, I'm working with SRP in mind so I'm going with your second suggestion to prevent a miscellaneous method name. If we need to do more on onbeforeunload, we should register more handlers that handle just that context.